### DISCUSSION TITLE
What is the unit/dimension of strokeWidth in Style object?

### DISCUSSION THREAD

**Discussion by eddiemuc** [mapsforge/vtm#1022] [Q&A] [answered]
I am using VTM on Android and have a very basic question wrt using styles in code.

Class `org.oscim.layers.vector.geometries.Style` has a field `strokeWidth` which takes a `float`. My question: what is the **unit/dimenstion** of this value? Is it pixels/points, dp (density-independent pixels) or an absolute value (meters, mm, km, ...)? Or something else entirely?

See here for a dimension description in the Android world:
https://developer.android.com/guide/topics/resources/more-resources#Dimension 

Apologies if this is a dumb question or easily explained somewhere in documentation, but I couldn't find it.

**Comment by devemux86** [comment, marked as answer]
On Android it will use the [Paint.setStrokeWidth(float)](https://developer.android.com/reference/android/graphics/Paint#setStrokeWidth(float)).
Their docs do not mention units (except hairline mode).

[Themes](https://github.com/mapsforge/vtm/blob/master/vtm/src/org/oscim/theme/XmlThemeBuilder.java#L567) and overlays scale such values with device density before use.
(similar to Mapsforge graphics)
